src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -115,7 +115,9 @@ public override void VisitSwitchStatement(SwitchStatementSyntax node)
 
         public override void VisitDeclarationPattern(DeclarationPatternSyntax node)
         {
-            _localsBuilder.Add(SourceLocalSymbol.MakeLocal(_scopeBinder.ContainingMemberOrLambda, _scopeBinder, false, node.Type, node.Identifier, LocalDeclarationKind.PatternVariable));
+            _localsBuilder.Add(SourceLocalSymbol.MakePatternLocalSymbol(
+                _scopeBinder.ContainingMemberOrLambda, _scopeBinder, _enclosingBinderOpt ?? _scopeBinder, node));
```

Comment:
>_enclosingBinderOpt ?? _scopeBinder [](start = 69, length = 35)

It looks like none of the existing tests verify that this change is needed for correctness and would prevent behavior from regressing.